src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,92 +1,101 @@
+//  * value is passed (typically in non-Typescript scenarios), then the default
+//  * is the "primary" variant.
+const getVariant = (variant, collection, fallback = null) => {
+  // Used to map between enum values and Chakra variant options.
+  const variantMap = {};
+  for (const type in collection) {
+    variantMap[collection[type]] = collection[type];
+  }
+  return variantMap[variant] || fallback;
+};
 
-function isTextBreadcrumb(obj: breadcrumb | JSX.Element): obj is breadcrumb {
-  return typeof (obj as breadcrumb).url === "string";
+export interface BreadcrumbstDataProps {
+  url: string;
+  text: string | React.ReactNode;
 }
 
 export interface BreadcrumbProps {
-  /** BlockName for use with BEM. See how to work with blockNames and BEM here: http://getbem.com/introduction/ */
-  blockName?: string;
   /** Breadcrumb links as an array */
-  breadcrumbs: breadcrumb[] | JSX.Element[];
+  breadcrumbsData: BreadcrumbstDataProps[];
   /** className you can add in addition to 'input' */
   className?: string;
-  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
-  modifiers?: string[];
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
+  /** Used to control how the `Hero` component will be rendered. */
+  colorVariant?: ColorVariants;
 }
 
-export default class Breadcrumbs extends React.Component<BreadcrumbProps, any> {
-  static defaultProps = {};
+const getElementsFromData = (data, breadcrumbsID) => {
+  const breadcrumbItems = [];
 
-  constructor(props: BreadcrumbProps) {
-    super(props);
+  if (!data?.length) {
+    return {};
   }
 
-  render(): JSX.Element {
-    const { breadcrumbs, className, blockName, modifiers } = this.props;
-
-    const baseClass = "breadcrumbs";
-
-    const breadcrumbItems = [];
-
-    if (!breadcrumbs || breadcrumbs.length === 0) {
-      throw new Error(
-        "Breadcrumbs must contain a set of links. Breadcrumbs currently empty"
-      );
-    }
-    breadcrumbs.forEach((item: breadcrumb | JSX.Element, index: number) => {
-      const last = index === breadcrumbs.length - 1;
-      let linkComponent: string | JSX.Element;
-
-      if (isTextBreadcrumb(item)) {
-        linkComponent = item.url ? (
-          <a href={item.url} className={bem("link", [], baseClass)}>
-            {item.text}
-          </a>
-        ) : (
-          item.text
-        );
-      } else {
-        const props = {
-          className: bem("link", [], baseClass),
-        };
-        linkComponent = React.createElement(
-          item.type,
-          { ...props, ...item.props },
-          item.props.children
-        );
-      }
-      breadcrumbItems.push(
-        <li
-          key={`${baseClass}-${index}`}
-          className={bem("item", [], baseClass)}
-        >
-          {last && (
+  data.map((breadcrumbData, index) => {
```

Comment:
This is complaining because `.map` returns an array. Instead of doing `const breadcrumbItems = []` and `breacrumbItems.push(...)`, directly return the JSX element and assign the variable to `.map`.

```jsx
const breadcrumbItems = data.map((...) => ({
  <BreadcrumbItem ...>
    ...
  </BreadcrumbItem>
});
```

Reply:
Ah, yes, more direct and simpler.  Thanks.